### ISSUE TITLE
"Cannot find namespace" errors ONLY in watch mode

### ISSUE THREAD

**Issue by chmln** [Microsoft/TypeScript#22257]
**TypeScript Version:**  2.8.0-dev.20180228

**Search Terms:** 
- cannot find namespace
- watch errors

**Repro**
1. `git clone https://github.com/flatpickr/flatpickr.git`
2. `yarn add typescript@next --dev`
3. `yarn test:typecheck`
4. Then compare with `yarn test:typecheck --watch`

**Expected behavior:**
If tsc does not find any errors, neither should `tsc watch`.

**Actual behavior:**
`tsc --watch` finds cryptic errors that are not present when running `tsc`.

**Playground Link:**  
Not applicable.

**Comment by chmln** [comment]
Figured it out after spending a few more hours.
Setting target + module to `esnext` and removing the `lib` option from tsconfig fixed the errors.

**Comment by mhegazy** [comment]
I am unable to get these errors locally.. i see no errors with or without `--watch`

**Comment by chmln** [comment]
@mhegazy that's because I just fixed the errors with a type assertion.

`export default fp.l10ns as FlatpickrFn["l10ns"];`

Would you be able to shed a light on why a type assertion be necessary here?

**Comment by mhegazy** [comment]
i see it now. thanks. it is a bug. the cast should not be needed.

**Comment by sheetalkamat** [comment]
Looks like the error comes through declaration emit: And same error is reported when executing
```node c:\TypeScript\built\local\tsc.js -p ./  --declaration``` and coming from doing typeToString...
